packages/calypso-e2e/src/lib/components/editor-component.ts

```diff
@@ -0,0 +1,99 @@
+import { Page, Locator } from 'playwright';
+
+const EDITOR_TIMEOUT = 60 * 1000;
+
+/**
+ * Represents the Editor component.
+ */
+export class EditorComponent {
+	private page: Page;
+	private parentLocator: Locator | null;
+	private canvasLocator: Locator | null;
+
+	/** */
+	constructor( page: Page ) {
+		this.page = page;
+		this.parentLocator = null;
+		this.canvasLocator = null;
+	}
+
+	/**
+	 * Returns the Editor parent locator. It will automatically resolve to the
+	 * proper locator, regardless if the Editor is Gutenframed or not.
+	 */
+	async parent(): Promise< Locator > {
+		if ( this.parentLocator ) {
+			return this.parentLocator;
+		}
+
+		try {
+			this.parentLocator = await Promise.any( [
```

Comment:
**comment** - when did we start being able to use `Promise.any`!? When I last tried to use it, VSCodium rejected my attempts. This is great to see!